pkg/controller/provider/ova-setup.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	err = r.createServerService(provider, ctx, ownerReference, labels)
+	if err != nil {
+		r.Log.Error(err, "Failed to create OVA server service")
+		return
+	}
+}
+
+func (r *Reconciler) createPvForNfs(provider *api.Provider, ctx context.Context, ownerReference metav1.OwnerReference, pvName string) (err error) {
+	splitted := strings.Split(provider.Spec.URL, ":")
+	if len(splitted) != 2 {
+		r.Log.Error(nil, "NFS server path doesn't contains: ", "provider", provider, "url", provider.Spec.URL)
+		return fmt.Errorf("wrong NFS server path")
+	}
+	nfsServer := splitted[0]
+	nfsPath := splitted[1]
+	labels := map[string]string{"providerName": provider.Name, "app": "forklift"}
+
+	pv := &core.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pvName,
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+			Labels:          labels,
+		},
+		Spec: core.PersistentVolumeSpec{
+			Capacity: core.ResourceList{
+				core.ResourceStorage: resource.MustParse(pvSize),
+			},
+			AccessModes: []core.PersistentVolumeAccessMode{
+				core.ReadOnlyMany,
+			},
+			PersistentVolumeSource: core.PersistentVolumeSource{
+				NFS: &core.NFSVolumeSource{
+					Path:   nfsPath,
+					Server: nfsServer,
+				},
+			},
+		},
+	}
+	err = r.Create(ctx, pv)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (r *Reconciler) createPvcForNfs(provider *api.Provider, ctx context.Context, ownerReference metav1.OwnerReference, pvName, pvcName string) (err error) {
+	sc := ""
+	labels := map[string]string{"providerName": provider.Name, "app": "forklift"}
+	pvc := &core.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pvcName,
+			Namespace:       provider.Namespace,
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+			Labels:          labels,
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{
+					core.ResourceStorage: resource.MustParse(pvSize),
+				},
+			},
+			AccessModes: []core.PersistentVolumeAccessMode{
+				core.ReadOnlyMany,
+			},
+			VolumeName:       pvName,
+			StorageClassName: &sc,
+		},
+	}
+	err = r.Create(ctx, pvc)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (r *Reconciler) createServerDeployment(provider *api.Provider, ctx context.Context, ownerReference metav1.OwnerReference, pvcName string, labels map[string]string) (err error) {
+	deploymentName := fmt.Sprintf("%s-deployment-%s", ovaServerPrefix, provider.Name)
+	annotations := make(map[string]string)
+	var replicas int32 = 1
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            deploymentName,
+			Namespace:       provider.Namespace,
+			Annotations:     annotations,
+			Labels:          labels,
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "forklift",
```

Comment:
shouldn't we extend this matcher - if we have another pod with the `app: forklift` label, we may think that we reached the number of desired replicas, no?